src/praisonai/praisonai/api/agent_invoke.py

```
@@ -339,16 +378,18 @@ async def invoke_agent(
                 detail=f"Agent '{agent_id}' not found"
             )
         
+        # Apply per-request agent_config overrides to the isolated clone, or
+        # reject unsupported keys with a 400 so the caller isn't silently
+        # running a different config than they asked for.
+        try:
+            _apply_agent_config(agent, request.agent_config)
+        except AgentConfigError as e:
+            raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)